### PR TITLE
- add rc orin for dcs 2.0

### PR DESCRIPTION
**Pull request by MartinSedlacek**
Added RC for orin_nx dcs 2.0.
- updated script and added new configuration

### CODE REVIEW

**Comment by MartinSedlacek** [comment]
This is postponed for a while. This works with at least one time flashed Orin NX, but it is not working with brand new one due to not flashing internals.

File: dcs_deploy.py [hunk 1]
```diff
@@ -666,7 +689,14 @@ def flash(self):
         print("Flash images! ...")
         self.prepare_status.change_group("flash")
         self.prepare_status.set_processing_step("flash_only")
-        ret = cmd_exec(f"sudo {self.flash_script_path} --flash-only {self.board_name} {self.rootdev}", print_command=True)
+        if self.config['device'] == 'xavier_nx':
+            ret = cmd_exec(f"sudo {self.flash_script_path} --flash-only {self.board_name} {self.qspi_partition_layout} {self.rootdev} {self.end_of_line}", print_command=True)
+            pass
```
Comment (eMrazSVK) [inline dcs_deploy.py:694]: This might be forgotten `pass` here.
```suggestion
```

File: dcs_deploy.py [hunk 2]
```diff
@@ -640,16 +656,23 @@ def generate_images(self):
...
                 env_vars = "ROOTFS_AB=1"
                 opt_app_size = "-S 4GiB "
                 external_only = "" # flash internal and external device
                 #self.rootdev = "external" # set UUID device in kernel commandline: rootfs=PARTUUID=<external-uuid>
-            ret = cmd_exec(f"sudo {env_vars} {self.flash_script_path} {opt_app_size} --no-flash {external_only} --external-device nvme0n1p1 " +
-                           f"-c {self.ext_partition_layout} --showlogs {self.board_name} {self.rootdev}", print_command=True)
+            
+            if self.config['device'] == 'xavier_nx':
+                ret = cmd_exec(f"sudo {env_vars} {self.flash_script_path} {opt_app_size} --no-flash {external_only} --external-device nvme0n1p1 --showlogs {self.board_name} {self.rootdev}", print_command=True)
+                pass
```
Comment (eMrazSVK) [inline dcs_deploy.py:668]: Unnecessary pass
```suggestion
```

File: dcs_deploy.py [hunk 3]
```diff
@@ -292,7 +292,7 @@ def get_download_file_path(self, url:str) -> str:
 
     def cleanup_old_download_dir(self):
         old_download_dir = self.config['device'] + '_' + self.config['storage'] + '_' + self.config['board'] + '_'
-
+
```
Comment (eMrazSVK) [inline dcs_deploy.py:295]:
```suggestion

```

File: dcs_deploy.py [hunk 4]
```diff
@@ -486,7 +498,7 @@ def prepare_sources_production(self):
         if self.get_resource_url('nv_ota_tools') != None:
             print('Applying Nvidia OTA tools ...')
             ret = self.extract_resource('nv_ota_tools')
-
+
```
Comment (eMrazSVK) [inline dcs_deploy.py:501]:
```suggestion

```

File: dcs_deploy.py [hunk 5]
```diff
@@ -501,9 +513,11 @@ def install_first_boot_setup(self):
         """
         Installs script that would be run on a device after the
         very first boot.
-        """
+        """
```
Comment (eMrazSVK) [inline dcs_deploy.py:516]:
```suggestion
        """
```

File: dcs_deploy.py [hunk 6]
```diff
@@ -640,16 +656,23 @@ def generate_images(self):
...
+                ret = cmd_exec(f"sudo {env_vars} {self.flash_script_path} {opt_app_size} --no-flash {external_only} --external-device nvme0n1p1 --showlogs {self.board_name} {self.rootdev}", print_command=True)
+                pass
+            elif self.config['device'] == 'orin_nx':   
+                os.chdir(self.l4t_root_dir)
+                ret = cmd_exec(f"sudo ./tools/kernel_flash/l4t_initrd_flash.sh --external-device nvme0n1p1  -c tools/kernel_flash/flash_l4t_external_custom.xml -p \"-c bootloader/t186ref/cfg/flash_t234_qspi.xml\" --no-flash --showlogs --network usb0 airvolute-dcs2.0+p3767-0000 internal", print_command=True)
+                print('sudo ./tools/kernel_flash/l4t_initrd_flash.sh --external-device nvme0n1p1  -c tools/kernel_flash/flash_l4t_external_custom.xml -p \"-c bootloader/t186ref/cfg/flash_t234_qspi.xml\"  --showlogs --network usb0 airvolute-dcs2.0+p3767-0000 internal')
+                
         self.prepare_status.set_status(ret, last_step= True)
-        return ret
+        return ret
```
Comment (eMrazSVK) [inline dcs_deploy.py:675]:
```suggestion
        return ret
```

**Review by eMrazSVK — CHANGES_REQUESTED** [review]
Basically just formatting changes. Other than that, I'm good with merging this one.

**Pull request closed without merging**